trikot-viewmodels-declarative-flow/compose-flow/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/viewmodel/material3/VMDBasicTextField.kt

```diff
@@ -0,0 +1,102 @@
+package com.mirego.trikot.viewmodels.declarative.compose.viewmodel.material3
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.LocalTextStyle
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextFieldColors
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.graphics.takeOrElse
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextStyle
+import com.mirego.trikot.viewmodels.declarative.components.VMDTextFieldViewModel
+import com.mirego.trikot.viewmodels.declarative.compose.extensions.composeValue
+import com.mirego.trikot.viewmodels.declarative.compose.extensions.isOverridingAlpha
+import com.mirego.trikot.viewmodels.declarative.compose.extensions.observeAsState
+import com.mirego.trikot.viewmodels.declarative.compose.extensions.vmdModifier
+import com.mirego.trikot.viewmodels.declarative.compose.viewmodel.internal.FormattedVisualTransformation
+
+@OptIn(ExperimentalMaterial3Api::class)
```

Comment:
Instead of Opt-in, we should propagate ExperimentalMaterial3Api annotation
i.e.
`@ExperimentalMaterial3Api`

same for the others

after that, the sample can opt-in at the gradle level.